test/functional/feature_rbf.py

```diff
@@ -622,5 +625,31 @@ def test_replacement_relay_fee(self):
         tx.vout[0].nValue -= 1
         assert_raises_rpc_error(-26, "insufficient fee", self.nodes[0].sendrawtransaction, tx.serialize().hex())
 
+    def test_fullrbf(self):
+        confirmed_utxo = self.wallet.get_utxo()
+
+        self.restart_node(0, extra_args=["-fullrbf"])
+
+        # Create an explictily opt-out transaction
+        optout_tx = self.wallet.send_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=confirmed_utxo,
+            sequence=SEQUENCE_FINAL,
+            fee_rate=Decimal('0.01'),
+        )
+        assert_equal(False, self.nodes[0].getmempoolentry(optout_tx['txid'])['bip125-replaceable'])
```

Comment:
This test assertion is failing (`AssertionError: not(False == True)`) for me.

Reply:
Same in CI: 
```
 2022-06-13T01:18:21.530000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 133, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_rbf.py", line 98, in run_test
                                       self.test_fullrbf()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_rbf.py", line 734, in test_fullrbf
                                       assert_equal(False, self.nodes[0].getmempoolentry(optout_tx['txid'])['bip125-replaceable'])
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 51, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(False == True)
```

Reply:
Test failure can be explained by the following reasoning.

If the outpoint is not confirmed, the parent transaction opt-in flags is going to be considered by our RPC code.
This is discrepancy between our RPC and mempool logic, see #22698.
Ensuring the spent outpoint is always confirmed should solve the issue, as only explicit signaling is considered in that case.